uui/components/pickers/__tests__/PickerInput.test.tsx

```diff
@@ -121,51 +161,921 @@ describe('PickerInput', () => {
+    });
+    
+    it('should render icon', async () => {
+        await setupPickerInputForTest({
+            value: undefined,
+            icon: () => <div data-testid = "icon" />,
+        });
+        
+        const target = screen.queryByTestId(/uui-PickerInput-target/);
+        const icon = within(target as HTMLElement).queryByTestId(/uui-PickerToggler-iconContainer/);
+        expect(icon).toBeDefined();
+    });
+    
+    it.each<['left' | undefined]>(
+        [[undefined], ['left']],
+    )('should render icon to the left', async (iconPosition) => {
+        await setupPickerInputForTest({
+            value: undefined,
+            icon: () => <div />,
+            iconPosition,
+        });
+        
+        const target = screen.queryByTestId(/uui-PickerInput-target/);
+        const icon = within(target as HTMLElement).queryByTestId(/uui-PickerToggler-iconContainer/);
+        expect(icon).toBeDefined();
+        
+        const elements = within(target as HTMLElement).queryAllByTestId(/uui-PickerToggler-/);
+        expect(elements.length).toBe(3);
+        expect(elements[1]).toEqual(icon);
+        
+        expect(elements[2]).toEqual(within(target as HTMLElement).queryByTestId(/uui-PickerToggler-input/));
+    });
+    
+    it('should render icon to the right', async () => {
+        await setupPickerInputForTest({
+            value: undefined,
+            icon: () => <div />,
+            iconPosition: 'right',
+        });
+        
+        const target = screen.queryByTestId(/uui-PickerInput-target/);
+        const icon = within(target as HTMLElement).queryByTestId(/uui-PickerToggler-iconContainer/);
+        expect(icon).toBeDefined();
+        
+        const elements = within(target as HTMLElement).queryAllByTestId(/uui-PickerToggler-/);
+        expect(elements.length).toBe(3);
+        expect(elements[1]).toEqual(within(target as HTMLElement).queryByTestId(/uui-PickerToggler-input/));
+        expect(elements[2]).toEqual(icon);
+    });
+
+    it('should pass onClick to the icon', async () => {
+        const onIconClick = jest.fn();
+        await setupPickerInputForTest({
+            value: undefined,
+            onIconClick,
+            icon: () => <div />,
+        });
+        
+        const target = screen.queryByTestId(/uui-PickerInput-target/);
+        const icon = within(target as HTMLElement).queryByTestId(/uui-PickerToggler-iconContainer/);
+        expect(icon).toBeDefined();
+
+        fireEvent.click(icon as HTMLElement);
+        expect(onIconClick).toBeCalledTimes(1);
+    });
+    
+    it('should open dialog only when minCharsToSearch is reached', async () => {
+        const { dom } = await setupPickerInputForTest({
+            value: undefined,
+            minCharsToSearch: 1,
+        });
+
+        fireEvent.click(dom.input as Element);
+
+        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+        
+        act(() => {
+            jest.useFakeTimers();
```

Comment:
maybe we could use findBy* queries to wait for apperance?

Reply:
Good point, thanks.